From 2c2771165d539ad22da410f4e006c0788a914298 Mon Sep 17 00:00:00 2001
From: Hui Liu <r64343@sdclx4.(none)>
Date: Thu, 13 Mar 2008 16:49:00 +0800
Subject: [PATCH] ENGR00068508 mtd nand bad block scan for large page

The mx32 ads board will crash when running at
the mxc nand scan bbt for large page nand flash

Signed-off-by: Jason Liu <r64343@freescale.com>
---
 drivers/mtd/nand/mxc_nd2.h |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/mtd/nand/mxc_nd2.h b/drivers/mtd/nand/mxc_nd2.h
index 5372d17..de661a6 100644
--- a/drivers/mtd/nand/mxc_nd2.h
+++ b/drivers/mtd/nand/mxc_nd2.h
@@ -141,8 +141,7 @@
 #define IS_4BIT_ECC			1
 #define SET_SPAS(v)
 #define SET_ECC_MODE(v)
-#define SET_NFMS(v)	\
-	raw_write((raw_read(NFMS) | (v)),NFMS)
+#define SET_NFMS(v)	(NFMS |= (v));
 
 #define WRITE_NFC_IP_REG(val,reg) \
 	raw_write((raw_read(REG_NFC_OPS_STAT) & ~NFC_OPS_STAT),  REG_NFC_OPS_STAT)
@@ -302,8 +301,7 @@
 #define IS_4BIT_ECC       1
 #define SET_SPAS(v)
 #define SET_ECC_MODE(v)
-#define SET_NFMS(v)	\
-	raw_write((raw_read(NFMS) | (v)),NFMS)
+#define SET_NFMS(v)	(NFMS |= (v));
 
 #define WRITE_NFC_IP_REG(val,reg) \
 	raw_write((raw_read(REG_NFC_OPS_STAT) & ~NFC_OPS_STAT),  REG_NFC_OPS_STAT)
-- 
1.5.4.4

